pkg/ipcache/metadata.go

```diff
@@ -472,7 +483,7 @@ func (ipc *IPCache) doInjectLabels(ctx context.Context, modifiedPrefixes []netip
 		}
 	}
 
-	for _, id := range previouslyAllocatedIdentities {
+	for prefix, id := range previouslyAllocatedIdentities {
```

Comment:
I might be missing something, but we never add ids with `id.createdFromMetadata == false` to `previouslyAllocatedIdentities` - thus I believe the force-release below is never triggered?

Reply:
Ah, right :facepalm: I think that one I missed a couple of times already.

Thanks for clarifying, looks good to me!